ec2utils/ec2utilsbase/lib/ec2utils/ec2utilsutils.py

```diff
@@ -175,21 +174,26 @@ def get_from_config(account, config, region, entry, cmd_line_arg):
 
 
 # -----------------------------------------------------------------------------
-def get_regions(command_args):
+def get_regions(command_args, access_key, secret_key):
     """Return a list of connected regions if no regions are specified
        on the command line"""
     regions = None
     if command_args.regions:
         regions = command_args.regions.split(',')
     else:
         regions = []
-        regs = boto.ec2.regions()
+        regs = boto3.client(
+                    aws_access_key_id=access_key,
+                    aws_secret_access_key=secret_key,
+                    region_name='us-east-1',
+                    service_name='ec2').describe_regions()['Regions']
```

Comment:
```
return boto3.session.Session(
    aws_access_key_id=access_key,
    aws_secret_access_key=secret_key
).get_available_regions('ec2')
```
This doesn't include disconnected regions by default, so no need to filter the out after ;-)

http://boto3.readthedocs.org/en/latest/reference/core/session.html?highlight=region#boto3.session.Session.get_available_regions